File: 238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.py
# Last updated: 7/2/2025, 11:53:02 PM
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        leftArr = [1]*(n)
        for i in range(1,n):
            leftArr[i] = leftArr[i-1]*nums[i-1]
        right = 1
        for i in range(n-1,-1,-1):
            leftArr[i] = leftArr[i]*right
            right = right*nums[i]
        return leftArr
        

File: 238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.cpp
// Last updated: 7/2/2025, 11:53:06 PM
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> ans;
        ans.push_back(1);
        int right=1;

        for(int i=1;i<nums.size();i++)
        {
            ans.push_back(ans[i-1]*nums[i-1]);
        }

        for(int i=nums.size()-1;i>=0;i--)
        {
            ans[i] = ans[i]*right;
            right = right*nums[i];
        }

        return ans;
        
    }
};